src/supermarket/spec/helpers/cookbook_versions_helper_spec.rb

```diff
@@ -24,4 +24,48 @@
       expect(helper.safe_updated_at(nil)).to be <= Time.zone.now
     end
   end
+
+  describe 'chef_versions_string' do
+    let(:chef_versions) { [['12.4.1', '12.4.2'], ['11.2.3', '12.4.3']] }
```

Comment:
I think each of these example versions ought to include an operator for consistency with the expect usage of the version constraints. 

I agree that the implementation is correct, though. I tossed the following into a working copy of this PR and things were green:

```ruby
    context 'with some operators' do
      it 'handles a simple range' do
        versions = [ [">= 11.14.5", "< 12.0.0"] ]
        expected_output = "(>= 11.14.5 AND < 12.0.0)"
        expect(helper.chef_versions_string(versions)).to eq(expected_output)
      end

      it 'handles what Lamont says are gnarly ranges' do
        versions = [
          [ ">= 11.14.5", "< 12.0.0" ],
          [ ">= 12.4.2", "< 13.0.0" ]
        ]
        expected_output = "(>= 11.14.5 AND < 12.0.0) OR (>= 12.4.2 AND < 13.0.0)"
        expect(helper.chef_versions_string(versions)).to eq(expected_output)
      end
    end
```